### PR TITLE
CLIP-1784: Multiple fixes to GitHub Actions

### PR DESCRIPTION
**Pull request by bianchi2**
Current problems:

* KinD tests fail when trigged by PRs created from forks because secrets are not available for actions triggered by `pull_request`.
* e2e tests triggered by labeled PRs checkout code from main branch in this repo rather than a fork

With this PR merged, to run KinD tests, `kind` label needs to be added to PR (or one can manually start this workflow). Also, e2e tests will checkout code from the fork and actually test the changes

## Checklist
- [ ] I have added unit tests
- [ ] I have applied the change to all applicable products
- [ ] The E2E test has passed (use `e2e` label)
- [ ] (Atlassian only) Internal Bamboo CI is passing

### CODE REVIEW

File: .github/workflows/e2e-tf-deployment.yaml [hunk 1]
```diff
@@ -40,9 +40,11 @@ jobs:
 
     steps:
       - name: Checkout Helm charts
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
           fetch-depth: 0
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          ref: ${{ github.head_ref }}
```
Comment (nanux) [inline .github/workflows/e2e-tf-deployment.yaml:47]: Did we really miss this? At some point, I believe we have verified that we are checking out the branch and testing the changes. Maybe it got lost in the ether.

Good catch.

File: src/test/config/kind/common-values.yaml [hunk 2]
```diff
@@ -21,23 +21,6 @@ DC_APP_REPLACEME:
   # this applies to Bamboo only and will be ignored in other Helm charts
   disableAgentAuth: true
 
-  # applicable to Bitbucket only
```
Comment (nanux) [inline src/test/config/kind/common-values.yaml:24]: Why we don't need this anymore?

File: src/test/scripts/kind/deploy_app.sh [hunk 3]
```diff
@@ -43,10 +43,7 @@ deploy_app() {
   fi
   sed -i "s/DC_APP_REPLACEME/${DC_APP}/g" ../../../test/config/kind/common-values.yaml
   sed -i "s/DB_TYPE_REPLACEME/${DB_TYPE}/g" ../../../test/config/kind/common-values.yaml
-  if [ ${DC_APP} == "bitbucket" ]; then
-    # this is to test mesh
-    IMAGE_OVERRIDE="--set image.tag=8.6.1"
```
Comment (nanux) [inline src/test/scripts/kind/deploy_app.sh:48]: Was this due to the mesh?

**Review by nanux — APPROVED** [review]

File: src/test/scripts/kind/deploy_app.sh (hunk 3 above)
Comment (bianchi2) [reply src/test/scripts/kind/deploy_app.sh:48]: Yes, leftovers. We test mesh in a separate Bamboo plan now.

File: .github/workflows/e2e-tf-deployment.yaml (hunk 1 above)
Comment (bianchi2) [reply .github/workflows/e2e-tf-deployment.yaml:47]: It was ok for branches in the repo, but not forks.

File: src/test/config/kind/common-values.yaml (hunk 2 above)
Comment (bianchi2) [reply src/test/config/kind/common-values.yaml:24]: That's mesh testing leftovers

**Pull request merged**